Fading collapsible

ChcJohnie opened the pull request:
closes #148 

Adds FadingCollapsible wrapping container which hides overflowing content unless expanded.
Used in Doc variant with defaults, utilized by /components installation block.
Also reused in TabbedCode component.

Note: as docs variant was not containing visually distinctive block (like with code), it feeled bit strange for me. I tried to mitigate with left side subtle border, so structure is more defined but Im defo open to other design suggestions.

ChcJohnie commented on src/docs/components/Code/TabbedCode.tsx:
```diff
@@ -17,100 +21,62 @@ export interface TabbedCodeItem {
 }: TabbedCodeProps) => {
-  const isExpandable = collapsedClassName || expandedClassName;
+  const isExpandable = typeof collapsedClassName === "string";
+  const Wrapper = isExpandable ? FadingCollapsible : React.Fragment;
```
This Wrapper related code is bit Vueish way of writing related logic. Is it okay in React?

iamhectorsosa replied:
Yes, this is also a valid pattern in React.

iamhectorsosa commented on src/docs/components/InstallationDocs/DocCollapsible.tsx:
```diff
@@ -0,0 +1,16 @@
+const defaultOptions: Omit<FadingCollapsibleProps, "children"> = {
+  expandButtonEl: "Show more",
+  collapseButtonEl: "Hide section",
+  className: "space-y-8 md:space-y-12 border-l border-stone-700 pl-2 md:pl-4",
```
This border left looks a bit off. I'd remove it and just leave the background gradient.

iamhectorsosa reviewed (commented):
@ChcJohnie this is exactly what I was looking for but we just need to hone some of the implementation details. Nice work!

iamhectorsosa commented:
I wrote a comment that wasn't included in the review. Please ping me for a quick call to discuss this. @ChcJohnie

ChcJohnie commented:
Applied some refactors discussed during call:
- renamed to CollapsibleContent
- refactored props (keeped same number but simplified naming/usage)
- adjusted in docs border (but keep free to refuse that 🙂 )
